syntax = "proto3";
import "common/rpc/base.proto";

service Language {
    // Performs synchronous ping of service
    rpc ping(Ping) returns (Pong);

    rpc languageEncode(LanguageEncodeRequest) returns (LanguageEncodeResponse);
    rpc languageDecode(LanguageDecodeRequest) returns (LanguageDecodeResponse);
    
}

message LanguageEncodeRequest {
    string user = 1;
    string text = 2;
}

message LanguageEncodeResponse {
    bool status = 1;
    optional string message = 2;
    repeated int32 tokens = 3;
}

message LanguageDecodeRequest {
    string user = 1;
    repeated int32 tokens = 3;
}

message LanguageDecodeResponse {
    bool status = 1;
    optional string message = 2;
    string text = 3;
}
